tests/director/api/package_api_test.go

```diff
@@ -241,3 +241,236 @@ func TestDocumentsInPackage(t *testing.T) {
+	output := graphql.PackageInstanceAuth{}
+
+	// WHEN
+	t.Log("Request package instance auth creation")
+	err = tc.RunOperation(ctx, pkgInstanceAuthCreationRequestReq, &output)
+
+	// THEN
+	require.NoError(t, err)
+	assertPackageInstanceAuth(t, pkgInstanceAuthRequestInput, output)
+
+	saveExample(t, pkgInstanceAuthCreationRequestReq.Query(), "request package instance auth creation")
+
+	// Fetch Application with packages
+	packagesForApplicationReq := fixPackagesRequest(application.ID)
+	pkgFromAPI := graphql.ApplicationExt{}
+
+	err = tc.RunOperation(ctx, packagesForApplicationReq, &pkgFromAPI)
+	require.NoError(t, err)
+
+	// Assert the package instance auths exists
+	require.Equal(t, 1, len(pkgFromAPI.Packages.Data))
+	require.Equal(t, 1, len(pkgFromAPI.Packages.Data[0].InstanceAuths))
+
+	// Fetch Application with package
+	instanceAuthID := pkgFromAPI.Packages.Data[0].InstanceAuths[0].ID
+	packagesForApplicationWithInstanceAuthReq := fixPackageWithInstanceAuthRequest(application.ID, pkg.ID, instanceAuthID)
+
+	err = tc.RunOperation(ctx, packagesForApplicationWithInstanceAuthReq, &pkgFromAPI)
+	require.NoError(t, err)
+
+	// Assert the package instance auth exist
+	require.Equal(t, instanceAuthID, pkgFromAPI.Package.InstanceAuth.ID)
+}
+
+func TestRequestPackageInstanceAuthDeletion(t *testing.T) {
+	ctx := context.Background()
+
+	application := registerApplication(t, ctx, "app-test-package")
+	defer unregisterApplication(t, application.ID)
+
+	pkg := createPackage(t, ctx, application.ID, "pkg-app-1")
+	defer deletePackage(t, ctx, pkg.ID)
+
+	pkgInstanceAuth := createPackageInstanceAuth(t, ctx, pkg.ID)
+
+	pkgInstanceAuthDeletionRequestReq := fixRequestPackageInstanceAuthDeletionRequest(pkgInstanceAuth.ID)
+	output := graphql.PackageInstanceAuth{}
+
+	// WHEN
+	t.Log("Request package instance auth deletion")
+	err := tc.RunOperation(ctx, pkgInstanceAuthDeletionRequestReq, &output)
+
+	// THEN
+	require.NoError(t, err)
+
+	saveExample(t, pkgInstanceAuthDeletionRequestReq.Query(), "request package instance auth deletion")
+}
+
+func TestSetPackageInstanceAuth(t *testing.T) {
+	ctx := context.Background()
+
+	application := registerApplication(t, ctx, "app-test-package")
+	defer unregisterApplication(t, application.ID)
+
+	pkg := createPackage(t, ctx, application.ID, "pkg-app-1")
+	defer deletePackage(t, ctx, pkg.ID)
+
+	pkgInstanceAuth := createPackageInstanceAuth(t, ctx, pkg.ID)
+
+	authInput := fixBasicAuth()
+	pkgInstanceAuthSetInput := fixPackageInstanceAuthSetInputSucceeded(authInput)
+	pkgInstanceAuthSetInputStr, err := tc.graphqlizer.PackageInstanceAuthSetInputToGQL(pkgInstanceAuthSetInput)
+	require.NoError(t, err)
+
+	setPackageInstanceAuthReq := fixSetPackageInstanceAuthRequest(pkgInstanceAuth.ID, pkgInstanceAuthSetInputStr)
+	output := graphql.PackageInstanceAuth{}
+
+	// WHEN
+	t.Log("Set package instance auth")
+	err = tc.RunOperation(ctx, setPackageInstanceAuthReq, &output)
+
```

Comment:
as in the previous comment, please test if the status is SUCCEEDED and the auth is provided.